aip/client-libraries/4235.md

```diff
@@ -0,0 +1,62 @@
+---
+id: 4235
+state: draft
+created: 2024-02-15
+---
+
+# Automatically populate fields in the request message
+
+For APIs that leverage request idempotency as described via [AIP-155], it could
+be useful to have the client libraries automatically populate the `request_id`
+if it is
+not already set by the customer.
+
+**Note:** This feature is primarily written for `request_id` fields within the
+request message. However, this feature should work for a field of any name, as
+long as the below conditions hold true.
```

Comment:
Is this note needed? Right now we only have guidance for the `request_id` field

Reply:
I think the aspirational side of me wants to say that we should keep this, because there is no reason why, if all of the conditions are met, we can't auto populate a field named something other than `request_id`. I would hope that most implementations aren't looking specifically for a `request_id` field by name, given that technically any field name could appear in the config - this can definitely be generalized software feature.

Reply:
Kept the note and changed the wording slightly based on the below comment